File: LeetCode/[M] - 71. Simplify Path/solution.go
package main

import "strings"

func simplifyPath(path string) string {
	var stack []string
	for _, way := range strings.Split(path, "/") {
		if way != "." && way != "" && way != ".." {
			stack = append(stack, way)
		} else if way == ".." && len(stack) != 0 {
			stack = stack[:len(stack)-1]
		}
	}
	return "/" + strings.Join(stack, "/")
}

File: LeetCode/[M] - 71. Simplify Path/solution.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        stack = []
        for way in path.split("/"):
            if way != "." and way != "" and way != "..":
                stack.append(way)
            elif way == ".." and stack:
                stack.pop()
        return "/" + "/".join(stack)
